### PR TITLE
feat: add ens avatar and name support

### PR DESCRIPTION
**Pull request by bigint**
image

### CODE REVIEW

**Comment by bigint** [comment]
Any updates on this @xnyl @noah-ren? 🙇🏻

File: src/components/utils/useENS.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  useEffect(() => {
+    async function resolveENS() {
+      if (address && ethers.utils.isAddress(address)) {
+        const ensName = await provider.lookupAddress(address);
+        if (ensName) setENSName(ensName);
+      }
+    }
+    resolveENS();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (xnyl) [inline src/components/utils/useENS.ts:19]: This rule should not be disabled. Maybe move `provider` to the module scope?

File: src/components/utils/useENS.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { ethers } from "ethers";
+import { env } from "../../constants/environmentVariables";
+import { useEffect, useState } from "react";
+
+export function useENS(address: string | null | undefined) {
+  const provider = new ethers.providers.JsonRpcProvider(
+    `https://mainnet.infura.io/v3/${env.INFURA_ID}`
+  );
+  const [ensName, setENSName] = useState<string | null>(null);
```
Comment (xnyl) [inline src/components/utils/useENS.ts:9]: We use convention like `[ensName, setEnsName]`

File: src/components/utils/useENS.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { ethers } from "ethers";
+import { env } from "../../constants/environmentVariables";
+import { useEffect, useState } from "react";
+
+export function useENS(address: string | null | undefined) {
+  const provider = new ethers.providers.JsonRpcProvider(
+    `https://mainnet.infura.io/v3/${env.INFURA_ID}`
```
Comment (xnyl) [inline src/components/utils/useENS.ts:7]: Make provider network dependant? `(mainnet|testnet).infura.io` ?

**Comment by xnyl** [comment]
> Any updates on this @xnyl @noah-ren? 🙇🏻

Hey bigint, nice work! We will for sure include it in the new major Bridge version. I left some comments, once resolved we can merge.

File: src/components/utils/useENS.ts (hunk 3 above)
Comment (bigint) [reply src/components/utils/useENS.ts]: @xnyl ENS in testnets are not stable and smart contracts are not audited, most of the site uses only in mainnet 🙇🏻  also its available only in rinkeby and ropsten

**Comment by bigint** [comment]
@xnyl Addressed your comments 🤗

**Comment by bigint** [comment]
@xnyl any updates ser?

**Comment by bigint** [comment]
Is the bridge project maintained? 😢

**Comment by xnyl** [comment]
> Is the bridge project maintained? cry

Yes it is maintained. We are focusing on the new major release which is Bridge with H2H support. Only critical fixes are being deployed for the current one. However, in recognition of your determination, we will deploy it this week :-)

**Pull request merged**